### PR TITLE
OGM-1147 IdSourceKey doesn't need to handle multi column types

### PR DESCRIPTION
**Pull request by Sanne**
- https://hibernate.atlassian.net/browse/OGM-1147

### CODE REVIEW

File: ehcache/src/main/java/org/hibernate/ogm/datastore/ehcache/persistencestrategy/kind/impl/SerializableIdSourceKey.java [hunk 1]
```diff
@@ -100,24 +109,25 @@ else if ( !table.equals( other.table ) ) {
...
+		return "SerializableIdSourceKey [columnName='" + columnName + "', columnValues='" + columnValue + "']";
 	}
 
 	@Override
 	public void writeExternal(ObjectOutput out) throws IOException {
 		out.writeInt( VERSION );
 		out.writeUTF( table );
-		out.writeObject( columnNames );
-		out.writeObject( columnValues );
+		//Wrapping in String[] and Object[] respectively as this used to be the format,
```
Comment (gunnarmorling) [inline ehcache/src/main/java/org/hibernate/ogm/datastore/ehcache/persistencestrategy/kind/impl/SerializableIdSourceKey.java]: Should we make this an option?

**Comment by gunnarmorling** [comment]
Looks great apart from that one question whether the compat. behavior for ISPN should be an option. I.e. allowing users to keep the legacy format for now but don't use it e.g. in new apps. In 6, this option would go away.

**Comment by Sanne** [comment]
Yes that's reasonable, but I'd rather open a follow up issue as it's not trivial - I'd need this change to move on with Hot Rod.

BTW the (future) encoding change would affect both Infinispan and Ehcache.

**Comment by Sanne** [comment]
merged!

**Comment by Sanne** [comment]
followed up by https://hibernate.atlassian.net/browse/OGM-1148

**Pull request merged**